packages/cli/src/commands/contract/compile.ts

```diff
@@ -24,84 +23,96 @@ export class CompileContract extends Command {
     release: Flags.boolean({
       default: false,
       char: "r",
-      description:
-        "A production contract should always be build in `release` mode for building optimized wasm",
+      description: "A production contract should always be build in `release` mode for building optimized wasm"
     }),
+    all: Flags.boolean({
+      default: false,
+      char: "a",
+      description: "Set all to true to compile all contracts"
+    })
   };
 
   static args = {
     contractName: Args.string({
       name: "contractName",
-      required: true,
+      required: false,
+      default: "",
       description: "Name of the contract to compile",
     }),
   };
 
   async run(): Promise<void> {
     const { args, flags } = await this.parse(CompileContract);
 
-    await ensureSwankyProject();
+    if (args.contractName == "" && !flags.all) {
+      this.error("No contracts were selected to compile")
+    }
 
+    await ensureSwankyProject();
     const config = await getSwankyConfig();
 
-    const contractInfo = config.contracts[args.contractName];
-    if (!contractInfo) {
-      this.error(`Cannot find a contract named ${args.contractName} in swanky.config.json`);
+    const contractNames = [];
+    if (flags.all) {
+      const contractList = readdirSync(path.resolve("contracts"), { withFileTypes: true });
```

Comment:
Aren't contracts listed in `swanky.config.json` ? Currently this logic leads to error, for example in case for example of empty directiory.

`Error: Cannot find contract info for test contract in swanky.config.json`

I think we should treat `swanky.config.json` as a source of list of all contracts.

Reply:
This would be as simple as 
```
      for (const contractName in config.contracts) {
        contractNames.push(contractName);
      }
```

Reply:
You're right. We need to decide what will be the final source of truth. This idea of reading the contract directory and comparing to config is an artefact from the past I think.
We'll address that in the upcoming feature that will allow to init a project with an existing contract and sync the config file according to the folder contents #111

Reply:
That's true... thank you so much! will fix it soon